droidlet/perception/semantic_parsing/utils/interaction_logger.py

```diff
@@ -13,6 +14,15 @@ def __init__(self, filepath):
             headers (list): List of string headers to be used in data store.
         """
         self.log_filepath = filepath
+        self.createdFile = False
+
+
+    def create_loggings_file(self, filepath):
+        self.createdFile = True
+
+        if not os.path.isfile(self.log_filepath):
+            with open(self.log_filepath, 'w') as f:
+                json.dump([], f, ensure_ascii=False, indent=4)
```

Comment:
You're checking that the log file does not exist, then create an empty file? I remember we discussed this, what I meant was, if the file exists, you load it into memory and append interactions. If not, write to it.

But I also think it's fine to just overwrite this file with the assumption that's you're saving one log file single session.

Reply:
This function gets called in logInteraction. Basically, if the file isn't created, create the file with an empty json list. This is important because if the file doesn't exist, reading the non-existent file as a json will crash. I will add comments/rewrite this method to make this more clear.

Reply:
> This function gets called in logInteraction. Basically, if the file isn't created, create the file with an empty json list. This is important because if the file doesn't exist, reading the non-existent file as a json will crash. I will add comments/rewrite this method to make this more clear.

What I mean is, you don't need to create an empty file here, you can just create it when writing. Anyway, this is not a big deal, if you want to keep it this way you can.